### PR TITLE
EDM-400: Add Enrollment config endpoint

### PR DESCRIPTION
**Pull request by rawagner**
After creating and approving CSR, you can use this new command to generate an enrollment config
 
Run `flightctl enrollmentconfig mycsr -p myeckey.pem` to get the enrollment-config file

### CODE REVIEW

**Comment by lkatalin** [comment]
Thank you @rawagner 

I think this may need to grow a `/{name}` field so that the user can get it for a specific device - like we have for the CSRs - since this will basically be a "last step" on top of the CSR steps. So it would match https://github.com/flightctl/flightctl/blob/main/api/v1alpha1/openapi.yaml#L1468 . Though this may imply some shared namespace  or naming convention between this API and the CSR API.

File: internal/service/enrollmentconfig.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+// (GET /api/v1/enrollmentconfig)
+func (h *ServiceHandler) EnrollmentConfig(ctx context.Context, request server.EnrollmentConfigRequestObject) (server.EnrollmentConfigResponseObject, error) {
+	cert, _, err := h.ca.Config.GetPEMBytes()
+	if err != nil {
+		return server.EnrollmentConfig400JSONResponse{Message: err.Error()}, nil
+	}
+	return server.EnrollmentConfig200JSONResponse{
+		EnrollmentService: v1alpha1.EnrollmentService{
+			Authentication: v1alpha1.EnrollmentServiceAuth{
+				ClientCertificateData: "",
```
Comment (lkatalin) [inline internal/service/enrollmentconfig.go:20]: something like
```
ClientCertificateData: h.store.CertificateSigningRequest().Get(ctx, store.NullOrgId, request.Name).ToApiResource().Status.Certificate,
```

**Comment by lkatalin** [comment]
Also would change NO-ISSUE to MGMT-18170

**Comment by lkatalin** [comment]
This version is working as expected for me with the CLI command. :rocket:

**Comment by rawagner** [comment]
thanks @lkatalin . I've updated the code as you suggested. The EnrollmentConfig endpoint now expects a name of approved CSR.

**Review by lkatalin — APPROVED** [review]

**Review by lkatalin — APPROVED** [review]

**Pull request merged**